python/ray/worker.py

```diff
@@ -248,6 +256,10 @@ def __init__(self):
         # CUDA_VISIBLE_DEVICES environment variable.
         self.original_gpu_ids = ray.utils.get_cuda_visible_devices()
 
+        self.lock = threading.Lock()
+        self.plasma_client_lock = threading.Lock()
+        self.local_scheduler_client_lock = threading.Lock()
```

Comment:
Do we need two locks?

Reply:
Probably could get away with one.

The idea was to lock IPCs to the individual local scheduler and plasma store. However, I believe it is not sufficient to lock individual IPCs. Instead, we have to lock blocks of code (which could include multiple IPCs) (which admittedly sounds like the kind of thing you shouldn't do).